### PR TITLE
Fix smb test

### PR DESCRIPTION
**Pull request by jingxu97**
This PR enable unit tests so that they can be triggered during
presubmit. It also fixes an issue in smb unit test.

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,7 +15,7 @@ BUILD_TOOLS_DIR = $(BUILD_DIR)/tools
 GO_ENV_VARS = GO111MODULE=on GOOS=windows
 
 .PHONY: all
-all: generate compile lint
```
Comment (kkmsft) [inline Makefile]: Is there a reason why we don't want to perform generate ? Wouldn't that result in folks checking in code without running generate ?

File: .prow.sh [hunk 2]
```diff
@@ -9,4 +9,4 @@
 
 # TODO: Once we add windows build rules and a Makefile, we can enable
```
Comment (kkmsft) [inline .prow.sh]: Is this TODO comment still required ?

File: Makefile (hunk 1 above)
Comment (jingxu97) [reply Makefile]: it does not work without installing protoc etc.  I am not sure whether we can install it easily.

File: .prow.sh (hunk 2 above)
Comment (jingxu97) [reply .prow.sh]: no, will remove it.

File: Makefile (hunk 1 above)
Comment (kkmsft) [reply Makefile]: I see. Is there a way for us to run specific make targets ?
If so we could run :
```
make compile
make test
```

and get a compile and run the test ?

File: integrationtests/smb_test.go [hunk 3]
```diff
@@ -7,7 +7,7 @@ import (
 	"math/rand"
 	"os"
 	"os/exec"
-        "strings"
```
Comment (kkmsft) [inline integrationtests/smb_test.go]: Since there are many test fixes in this, could we do it as separate PRs, so that we can merge one and work on the comments for another one ?

Comment (jingxu97) [reply integrationtests/smb_test.go]: I spitted it
https://github.com/kubernetes-csi/csi-proxy/pull/22
